guides/micronaut-cloud-oidc-oracle/micronaut-cloud-oidc-oracle.adoc

```diff
@@ -0,0 +1,178 @@
+external:micronaut-cloud-oidc-base/start.adoc[]
+
+common:oracle-cloud-cli.adoc[]
+
+common:oracle-cloud-account.adoc[]
+
+common:completesolution.adoc[]
+
+== Configure OpenID Connect at Oracle Cloud
+
+We'll use the Oracle Cloud web console to create an OAuth 2.0 "Confidential Application" with support for OpenID Connect and a federated user to demonstrate using OpenID Connect with Micronaut and Oracle Cloud.
+
+=== Create a federated user
+
+https://cloud.oracle.com/[Login to your Oracle Cloud tenancy] as an admin (or a user with sufficient permissions to create users and applications), and from the Oracle Cloud Menu select "Identity & Security" and then "Federation":
+
+image::cloud-oidc-oracle/federation01.png[]
+
+Click on the link for "OracleIdentityCloudService":
+
+image::cloud-oidc-oracle/federation02.png[]
+
+and then click the "Create User" button:
+
+image::cloud-oidc-oracle/federation03.png[]
+
+Enter a valid username, e.g., "mn_guide_oidc", and a valid email address, along with a first and last name, then click "Create":
+
+image::cloud-oidc-oracle/federation04.png[]
+
+Open the email sent to the email address you specified and click the link to set the password for the user:
+
+image::cloud-oidc-oracle/federation05.png[]
+
+Login as the user to verify the password, then log out and back in as the admin user.
+
+=== Create the OAuth application
+
+Navigate again to the "OracleIdentityCloudService" federation and copy the link to the "Oracle Identity Cloud Service Console", minus the path (it will look like `\https://idcs-7084de14ec.......identity.oraclecloud.com/`).
+Save this for later when we create a Micronaut application where this value will be used as the `OAUTH_ISSUER`.
+
+Click the "Oracle Identity Cloud Service Console" link to open the console:
+
+image::cloud-oidc-oracle/federation06.png[]
+
+Click the button in the top right of the "Applications and Services" section to create a new application:
+
+image::cloud-oidc-oracle/federation07.png[]
+
+and then click "Confidential Application":
+
+image::cloud-oidc-oracle/federation08.png[]
+
+Enter a valid application name, e.g., "mn_guide_oidc_app" and optionally a description:
+
+image::cloud-oidc-oracle/federation09.png[]
+
+Click "Next" to go to step 2.
+
+Click "Configure this application as a client now", and in the `Allowed Grant Types` section check `Client Credentials` and `Authorization Code`.
+
+Check `Allow non-HTTPS URLs` for now to make testing easier and avoid having to use HTTPS locally.
+Be sure to unselect this in production applications which should always use HTTPS.
+
+Enter `\http://localhost:8080/oauth/callback/oci` for "Redirect URL", `\http://localhost:8080/logout` for "Logout URL", and `\http://localhost:8080/` for "Post Logout Redirect URL":
```

Comment:
I think it might be easier for a user to follow these directions if they were split up as so:

```suggestion
Enter the following:

 - Redirect URL: `\http://localhost:8080/oauth/callback/oci` 
 - Logout URL: `\http://localhost:8080/logout`
 - Post Logout Redirect URL: `\http://localhost:8080/`
```